src/Microsoft.Health.Dicom.Core/Configs/RetrieveConfiguration.cs

```diff
@@ -0,0 +1,21 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Dicom.Core.Configs
+{
+    public class RetrieveConfiguration
+    {
+        /// <summary>
+        /// Maximum dicom file supported for getting frames and transcoding
+        /// </summary>
+        public long MaxDicomFileSize { get; } = 1024 * 1024 * 100; //100 MB
+
+        /// <summary>
+        /// Response uses lazy streams that copy data from storage JIT into a buffer and then to the output stream
+        /// This is the size of the buffer
+        /// </summary>
+        public int LazyResponseStreamBufferSize { get; } = 1024 * 1024 * 4; //4 MB
```

Comment:
+1 on the article. I did a quick digging, since Azure Blob SDK uses the same size.

Here is the source code of arrayPool https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs#L281

Looks like there are 16 buckets of varying sizes (16 bytes to 1MB) and each bucket has 8 buffers.

I also did a quick perfview check and Array pool created reason was PoolExhausted and not overMaximumSize. 

![image](https://user-images.githubusercontent.com/6980724/154816614-0a654e37-48e6-48f0-9da6-f9b56ceb5769.png)

Reply:
@smitha the buffer sizes in the logs above are all less than 1MB so they aren't testing the 4MB behavior. My concern is that the behavior of ArrayPool in the file you linked will not use ArrayPool at all - it will [check to find a bucket index](https://github.com/dotnet/corert/blob/c6af4cfc8b625851b91823d9be746c4f7abdc667/src/System.Private.CoreLib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs#L100) that fits 4MB.

Since the value returned (18) is not less than the number of buckets (17), [a new array will be allocated](https://github.com/dotnet/corert/blob/c6af4cfc8b625851b91823d9be746c4f7abdc667/src/System.Private.CoreLib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs#L143) and **not** returned to the pool.

Reply:
Here is another snapshot of 4MB, with the same poolExhausted. I have also asked the experts. We can always change it later if they come back with the better suggestion

![image](https://user-images.githubusercontent.com/6980724/155272790-d7fdf696-7f91-4c65-918b-3f113dd6dbf2.png)

Reply:
Just figured it out - the article I linked and the code you linked is from the `dotnet/corert` repo, which is now archived. That repo did have the maximum array size of 1MB, but there's a completely new repo for these libraries at `dotnet/runtime`. There they did increase the [maximum array size](https://github.com/dotnet/runtime/blob/2a9745176ac0cea4a86b37c52e9863d7d13f24fc/src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs#L24) to **1GB**. I ran the same perfview and was also seeing that 4MB was not over the maximum. Sorry about the confusion! 🙂